Sources/SwiftSyntax/TokenSyntax.swift

```diff
@@ -153,6 +153,11 @@ public struct TokenSyntax: SyntaxProtocol, SyntaxHashable {
     }
   }
 
+  /// An identifier created from `self`.
+  public var identifier: Identifier? {
+    Identifier(self)
```

Comment:
This should return `nil` if `self.tokenKind` is not `.identifier` or `.dollarIdentifier`.